File: April 2020/03 - Maximum Subarray/Solution.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSum = nums[0], winSum = nums[0];
        for(int i = 1; i < nums.size(); ++i) {
            winSum = max(winSum + nums[i], nums[i]);
            maxSum = max(maxSum, winSum);
        }        
        return maxSum;
    }
};

File: April 2020/03 - Maximum Subarray/Solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        windowSum = nums[0]
        maxSum = nums[0]
        for num in range(1,len(nums)):
            windowSum = max(windowSum + nums[num], nums[num])
            maxSum = max(maxSum, windowSum)
        return maxSum